tests/test_regrid.py

```diff
@@ -770,13 +772,144 @@ def test_preserve_bounds(self):
+
+        assert np.array_equal(axis, self.lat_data)
+        assert bnds is not None
+        assert np.array_equal(bnds, self.lat_bnds_data)
+
+    def test_create_axis_no_bnds(self):
+        expected_axis_attrs = {
+            "axis": "Y",
+            "units": "degrees_north",
+            "coordinate": "latitude",
+        }
+
+        axis, bnds = grid.create_axis("lat", self.lat_data, generate_bounds=False)
+
+        assert np.array_equal(axis, self.lat_data)
+        assert bnds is None
+        assert axis.attrs == expected_axis_attrs
+
+    def test_create_axis_user_bnds(self):
+        expected_axis_attrs = {
+            "axis": "Y",
+            "units": "degrees_north",
+            "coordinate": "latitude",
+            "bounds": "lat_bnds",
+        }
+
+        axis, bnds = grid.create_axis("lat", self.lat_data, bounds=self.lat_bnds_data)
+
+        assert np.array_equal(axis, self.lat_data)
+        assert bnds is not None
+        assert np.array_equal(bnds, self.lat_bnds_data)
+        assert "xcdat_bounds" not in bnds.attrs
+        assert axis.attrs == expected_axis_attrs
+
+    def test_create_axis_invalid_name(self):
+        with pytest.raises(
+            ValueError, match="The name 'mass' is not valid for an axis name."
+        ):
+            grid.create_axis("mass", self.lat_data)
+
     def test_empty_grid(self):
         with pytest.raises(
-            ValueError, match="Must pass at least 1 coordinate to create a grid."
+            ValueError, match="Must pass at least 1 axis to create a grid."
         ):
             grid.create_grid()
 
-    def test_unexpected_coordinate(self):
+    def test_create_grid(self):
+        new_grid = grid.create_grid(x=self.lon, y=self.lat)
+
+        assert np.array_equal(new_grid.lat, self.lat)
+        assert np.array_equal(new_grid.lon, self.lon)
+
+    def test_create_grid_with_bounds(self):
+        new_grid = grid.create_grid(
+            x=(self.lon, self.lon_bnds), y=(self.lat, self.lat_bnds)
+        )
+
+        assert np.array_equal(new_grid.lat, self.lat)
+        assert new_grid.lat.attrs["bounds"] == self.lat_bnds.name
+        assert np.array_equal(new_grid.lat_bnds, self.lat_bnds)
+
+        assert np.array_equal(new_grid.lon, self.lon)
+        assert new_grid.lon.attrs["bounds"] == self.lon_bnds.name
+        assert np.array_equal(new_grid.lon_bnds, self.lon_bnds)
+
+    def test_create_grid_user_attrs(self):
+        lev = xr.DataArray(np.linspace(1000, 1, 2), dims=["lev"], name="lev")
+
+        new_grid = grid.create_grid(z=lev, attrs={"custom": "value"})
+
+        assert "custom" in new_grid.attrs
+        assert new_grid.attrs["custom"] == "value"
+
+    def test_create_grid_wrong_axis_value(self):
+        with pytest.raises(
+            ValueError,
+            match=re.escape(
+                "Argument 'x' should either be a xr.DataArray or (xr.DataArray, xr.DataArray)"
```

Comment:
If the related code suggestion is accepted, this ValueError message also needs to be updated.